go/arrow/array/data.go

```diff
@@ -190,9 +190,32 @@ func (d *Data) SetDictionary(dict arrow.ArrayData) {
 	}
 }
 
+// Returns the size of Data in bytes
+func (d *Data) SizeInBytes() uint64 {
+	var size uint64
+
+	if d == nil {
+		return 0
+	}
+
+	for _, b := range d.Buffers() {
+		size += uint64(b.Len())
+	}
```

Comment:
should the documentation say that the return of `SizeInBytes` is an upper-bound since we don't take into account any offset?

The reason i say this is that taking into account the offset would actually be quite complex since the offset value is based on the element, not the raw bytes, and thus would make this computation significantly more complex. I think it's not particularly worthwhile to incorporate the offset into this and is good enough to say that the `SizeInBytes` is just an upper-bound if offset is 0.

Reply:
Make sense.
A quick question, will the calculation with/without offset differs a lot?

Reply:
It could potentially be very different depending on the difference between the offset and the size. If the offset makes it so the arrow array is only of size 1 (i.e. a slice of 1 from a very large array) then you could have a significant difference in the calculation